src/NServiceBus.Core/Recoverability/SecondLevelRetries/CustomSecondLevelRetryPolicy.cs

```diff
@@ -14,7 +14,7 @@ public override bool TryGetDelay(IncomingMessage message, Exception ex, int curr
         {
             delay = customRetryPolicy(message);
 
-            return delay != TimeSpan.Zero;
+            return delay != TimeSpan.MinValue;
```

Comment:
Ok I didn't know that. Sorry that was either never told to me or I forgot it (which is probably more likely)